### PR TITLE
fix: disable portfolio description if over 288 characters

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Ref #49132

### CODE REVIEW

**Comment by raisedadead** [comment]
Can you elaborate on the changes and the context in the OP when making a PR if you are not linking to a previous thread/issue?

Thanks.

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Comment by Sembauke** [comment]
> Can you elaborate on the changes and the context in the OP when making a PR if you are not linking to a previous thread/issue?
> 
> Thanks.

I thought it would be straightforward enough when you read the title.

**Comment by raisedadead** [comment]
Thanks for adding the details. 

Let's be mindful that the project is maintained and followed by people who may not be on all these meetings :)

**Comment by Sembauke** [comment]
You are right.

**Review by raisedadead — DISMISSED** [review]
The change looks good, but where is the error message that the description can't be longer than 288? 

Also, since this is a criterion, let's make it concrete by setting a const somewhere up top in the file and avoid the magic number + a test to go along with this, so it doesn't pop-up again.

**Comment by Sembauke** [comment]
> The change looks good, but where is the error message that the description can't be longer than 288?

This is already in place. Right under the textfield.

**Comment by moT01** [comment]
Update the branch to pass the "Localize" tests

**Comment by raisedadead** [comment]
> This is already in place. Right under the textfield.

Perfect. Thanks for confirming.

As soon as the other nit pick and Tom's comments are addressed, this should be good to go.

**Comment by Sembauke** [comment]
Hey @raisedadead 

Thanks for the requested changes, I have made the decision that putting the variable in the State interface was not a good idea (Making it a global variable in the beginning of the file), the state needs to be set before the React Component renders.  

In my opinion this is the most careful way of doing things, a.k.a. Making a function which looks for all the possible wronging that could happen set in one variable (isButtonDisabled), while allowing the other variables to be accessed.  

As for the tests, they check if it is possible to make a portfolio items, and check for the correct error handling if the input from the user is not valid in one of the textfields.

File: client/src/components/helpers/form/block-save-button.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ function BlockSaveButton({
 }: {
   children?: React.ReactNode;
   disabled?: boolean;
+  dataLabel?: string;
```
Comment (Sembauke) [inline client/src/components/helpers/form/block-save-button.tsx:12]:
```suggestion
```

**Comment by raisedadead** [comment]
Even better. I only wanted to avoid the magic number 288. But you seem to have found a way to refactor this, which works for me.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**